sources/packages/api/src/route-controllers/application/application.controller.service.ts

```diff
@@ -203,4 +211,98 @@ export class ApplicationControllerService {
       );
     }
   }
+
+  /**
+   * Process application income verification details.
+   * @param craIncomeVerification CRA income verification data.
+   * @return processed object.
+   */
+  processApplicationIncomeVerificationDetails(
+    craIncomeVerification: CRAIncomeVerification[],
+  ): ApplicationIncomeVerification {
+    const incomeVerificationDetails = {} as ApplicationIncomeVerification;
+    // Student.
+    const [student] = craIncomeVerification.filter(
+      (incomeVerification) => !incomeVerification.supportingUser,
+    );
+    if (student) {
+      // Student income verification details.
+      incomeVerificationDetails.studentIncomeVerificationStatus =
+        !student.dateReceived
+          ? SuccessWaitingStatus.Waiting
+          : SuccessWaitingStatus.Success;
+    }
+    // Parent.
+    const [parent1, parent2] = craIncomeVerification.filter(
+      (incomeVerification) =>
+        incomeVerification.supportingUser?.supportingUserType ===
+        SupportingUserType.Parent,
+    );
+    if (parent1) {
+      incomeVerificationDetails.parent1IncomeVerificationStatus =
+        !parent1.dateReceived
+          ? SuccessWaitingStatus.Waiting
+          : SuccessWaitingStatus.Success;
+    }
+    if (parent2) {
+      incomeVerificationDetails.parent2IncomeVerificationStatus =
+        !parent2.dateReceived
+          ? SuccessWaitingStatus.Waiting
+          : SuccessWaitingStatus.Success;
+    }
+
+    // Partner.
+    const [partner] = craIncomeVerification.filter(
+      (incomeVerification) =>
+        incomeVerification.supportingUser?.supportingUserType ===
+        SupportingUserType.Partner,
+    );
+    // partner income verification details.
+    if (partner) {
+      incomeVerificationDetails.partnerIncomeVerificationStatus =
+        !partner.dateReceived
+          ? SuccessWaitingStatus.Waiting
+          : SuccessWaitingStatus.Success;
+    }
+    return incomeVerificationDetails;
+  }
+
+  /**
+   * Process application supporting user details.
+   * @param supportingUser supporting users data.
+   * @return processed object.
+   */
+  processApplicationSupportingUserDetails(
+    supportingUser: SupportingUser[],
+  ): ApplicationSupportingUserDetails {
+    const supportingUserDetails = {} as ApplicationSupportingUserDetails;
+    // Parent.
+    const [parent1, parent2] = supportingUser.filter(
+      (incomeVerification) =>
+        incomeVerification.supportingUserType === SupportingUserType.Parent,
+    );
+    if (parent1) {
+      supportingUserDetails.parent1Info = !parent1.supportingData
+        ? SuccessWaitingStatus.Waiting
+        : SuccessWaitingStatus.Success;
+    }
+    if (parent2) {
+      supportingUserDetails.parent1Info = !parent2.supportingData
```

Comment:
Is it supposed to be `supportingUserDetails.parent2Info`, right?

Reply:
yes, thanks for that